dwave/optimization/src/nodes/lp.cpp

```diff
@@ -34,15 +34,18 @@ struct LPData {
     std::vector<double> ub;
 };
 
-struct LPNodeData : ArrayNodeStateData {
-    explicit LPNodeData(std::vector<double>&& values, bool is_feasible,
+struct LPNodeData : NodeStateData {
+    explicit LPNodeData(std::vector<double>&& solution, bool is_feasible,
                         double objective_value) noexcept
-            : ArrayNodeStateData(std::move(values)),
+            : solution(std::move(solution)),
               is_feasible(is_feasible),
               old_is_feasible(is_feasible),
               objective_value(objective_value),
               old_objective_value(objective_value) {}
 
+    std::vector<double> solution;
```

Comment:
I think we do want the `LPSolutionNode` to be statefull in this case, and to hold the solution.

Reply:
Refactored to eliminate this extra state. Had to remove one check in the unit tests that relied on the `LPNode::feasible()` being up to date which it no longer is guaranteed to be.